### PR TITLE
resolve two unused warnings in libtester that show up only in release builds

### PR DESCRIPTION
**Pull request by spoonincode**
clang18:
```
/home/spoon/f/leap/src/libraries/testing/tester.cpp:378:25: warning: unused variable '[block, id]' [-Wunused-variable]
  378 |             const auto& [block, id] = t;
      |                         ^
/home/spoon/f/leap/src/libraries/testing/tester.cpp:377:106: warning: lambda capture 'this' is not used [-Wunused-lambda-capture]
  377 |       [[maybe_unused]] auto accepted_block_header_connection = control->accepted_block_header().connect([this](const block_signal_params& t) {
      |                                                                                                          ^~~~
```

an alternative would be to wrap the 5 lines of this connection in a `#ifndef NDEBUG`

### CODE REVIEW

File: libraries/testing/tester.cpp [hunk 1]
```diff
@@ -374,8 +374,8 @@ namespace eosio::testing {
       [[maybe_unused]] auto block_start_connection = control->block_start().connect([](block_num_type num) {
          // only block number is signaled, in forking tests will get the same block number more than once.
       });
-      [[maybe_unused]] auto accepted_block_header_connection = control->accepted_block_header().connect([this](const block_signal_params& t) {
-            const auto& [block, id] = t;
+      [[maybe_unused]] auto accepted_block_header_connection = control->accepted_block_header().connect([&](const block_signal_params& t) {
```
Comment (heifner) [inline libraries/testing/tester.cpp:377]: Why the change from `[this]` to `[&]` ?

**Review by greg7mdp — APPROVED** [review]

File: libraries/testing/tester.cpp (hunk 1 above)
Comment (spoonincode) [reply libraries/testing/tester.cpp:377]: It gets rid of the second warning. `warning: lambda capture 'this' is not used`. I don't think we can use `[[maybe_unused]]` in capture list? If you have preference for another approach I'm totally open to it

**Review by heifner — APPROVED** [review]

**Comment by ericpassmore** [comment]
Note:start
category: Tests
component: Internal
summary: Resolve unused warning in libtester.
Note:end
